nemo/collections/asr/metrics/rnnt_wer.py

```diff
@@ -191,17 +191,23 @@ class AbstractRNNTDecoding(ConfidenceMixin):
         blank_id: The id of the RNNT blank token.
     """
 
-    def __init__(self, decoding_cfg, decoder, joint, blank_id: int):
+    def __init__(self, decoding_cfg, decoder, joint, blank_id: int, num_extra_outputs: int = 0):
```

Comment:
Instead of taking it as an argument, do joint.num_extra_outputs, and AbstractRNNTJoint will now have two properties - `num_classes_with_blank` and `num_extra_outputs` that will both return ints. By default the impl of these two properties is to raise ValueError().

Subclasses must implement these methods, calculating their vocab size and their num_extra_outputs. 

The decoding framework can then reliably access these values from the joint itself rather than accept such values via its constructor. The Decoding framework's constructor should only contain a few elements - the config, the decoder module, joint module and the blank id.

Reply:
Done.